common-controller/internal/controlplane/k8s_artifact_deployer.go

```diff
@@ -0,0 +1,333 @@
+	}
+	if keyMapping.SecurityScheme == "OAuth2" {
+		if securitySchemes.OAuth2 == nil {
+			securitySchemes.OAuth2 = &cpv1alpha2.SecurityScheme{Environments: []cpv1alpha2.Environment{generateSecurityScheme(keyMapping)}}
+		} else {
+			environments := make([]cpv1alpha2.Environment, 0)
+			for _, environment := range securitySchemes.OAuth2.Environments {
+				if environment.EnvID != keyMapping.EnvID || environment.AppID != keyMapping.ApplicationIdentifier || environment.KeyType != keyMapping.KeyType {
+					environments = append(environments, environment)
+				}
+			}
+			securitySchemes.OAuth2.Environments = append(environments, generateSecurityScheme(keyMapping))
+		}
+	}
+	crApplication.Spec.SecuritySchemes = &securitySchemes
+	loggers.LoggerAPKOperator.Infof("Updating Application %v", crApplication)
+	return k8sArtifactDeployer.client.Update(context.Background(), &crApplication)
+}
+
+// DeleteAllApplicationMappings deletes all application mappings
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteAllApplicationMappings() error {
+	return nil
+}
+
+// DeleteAllApplications deletes all applications
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteAllApplications() error {
+	return nil
+}
+
+// DeleteAllKeyMappings deletes all key mappings
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteAllKeyMappings() error {
+	return nil
+}
+
+// DeleteAllSubscriptions deletes all subscriptions
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteAllSubscriptions() error {
+	return nil
+}
+
+// DeleteApplication deletes an application
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteApplication(applicationID string) error {
+	crApplication := cpv1alpha2.Application{}
+	err := k8sArtifactDeployer.client.Get(context.Background(), client.ObjectKey{Name: applicationID, Namespace: utils.GetOperatorPodNamespace()}, &crApplication)
+	if err != nil {
+		if !k8error.IsNotFound(err) {
+			loggers.LoggerAPKOperator.ErrorC(logging.PrintError(logging.Error1102, logging.BLOCKER, "Failed to get application from k8s %v", err.Error()))
+			return err
+		}
+	} else {
+		err := k8sArtifactDeployer.client.Delete(context.Background(), &crApplication)
+		if err != nil {
+			loggers.LoggerAPKOperator.ErrorC(logging.PrintError(logging.Error1100, logging.BLOCKER, "Failed to delete application in k8s %v", err.Error()))
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeleteApplicationMappings deletes an application mapping
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteApplicationMappings(applicationMapping string) error {
+	crApplicationMapping := cpv1alpha2.ApplicationMapping{}
+	err := k8sArtifactDeployer.client.Get(context.Background(), client.ObjectKey{Name: applicationMapping, Namespace: utils.GetOperatorPodNamespace()}, &crApplicationMapping)
+	if err != nil {
+		if !k8error.IsNotFound(err) {
+			loggers.LoggerAPKOperator.ErrorC(logging.PrintError(logging.Error1102, logging.BLOCKER, "Failed to get application mapping from k8s %v", err.Error()))
```

Comment:
I think we don't want to restart the adapter if we encounter such errors. logging.BLOCKER will crash the adapter. Use it carefully

Reply:
Please fix in all places.